engines/agi/menu.cpp

```diff
@@ -82,26 +85,34 @@ void GfxMenu::addMenu(const char *menuText) {
 
 	menuEntry->textLen = menuEntry->text.size();
 
-	// Cut menu name in case menu bar is full
-	// Happens in at least the fan game Get Outta Space Quest
-	// Original interpreter had graphical issues in this case
-	// TODO: this whole code needs to get reworked anyway to support different types of menu bars depending on platform
-	curColumnEnd += menuEntry->textLen;
-	while ((menuEntry->textLen) && (curColumnEnd > 40)) {
-		menuEntry->text.deleteLastChar();
-		menuEntry->textLen--;
-		curColumnEnd--;
+	if (!_vm->isLanguageRTL()) {
+		// Cut menu name in case menu bar is full
+		// Happens in at least the fan game Get Outta Space Quest
+		// Original interpreter had graphical issues in this case
+		// TODO: this whole code needs to get reworked anyway to support different types of menu bars depending on platform
+		curColumnEnd += menuEntry->textLen;
+		while ((menuEntry->textLen) && (curColumnEnd > 40)) {
```

Comment:
Not your code, but should probably be changed from 40 to `FONT_COLUMN_CHARACTERS`

Reply:
Well, you're welcomed to open a PR with such improvements.
It's out of scope of this PR :-)